examples/vision/pointnet.py

```diff
@@ -0,0 +1,295 @@
+"""
+Title: Point cloud classification with PointNet
+Author: [David Griffiths](https://dgriffiths3.github.io)
+Date created: 2020/05/25
+Last modified: 2020/05/25
+Description: Implementation of PointNet for ModelNet10 classification.
+"""
+
+"""
+## Introduction
+"""
+
+"""
+Classification, detection and segmentation of unordered 3D point sets i.e. point clouds
+is a core problem in computer vision. This example implements the seminal point cloud
+deep learning paper [PointNet (Qi et al., 2017)](https://arxiv.org/abs/1612.00593). For a
+detailed intoduction on PointNet see [this blog
+post](https://medium.com/@luis_gonzales/an-in-depth-look-at-pointnet-111d7efdaa1a).
+"""
+
+"""
+## Setup
+"""
+
+"""
+If running in google colab first run `!pip install trimesh`.
+"""
+
+import glob
+import trimesh
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.keras import backend as K
+
+"""
+## Load dataset
+"""
+
+"""
+We use the ModelNet10 model dataset, the smaller 10 class version of the ModelNet40
+dataset. First download the data:
+"""
+
+"""shell
+!curl -O http://3dvision.princeton.edu/projects/2014/3DShapeNets/ModelNet10.zip
```

Comment:
Prefer using `get_file` to download and cache the file, and unzip it in Python. Having shell commands means the script can't run as a standalone Python file, so shell commands should only be used when there's no alternative.

Reply:
Actually, nevermind that comment, shell commands should be fine in this case.

Reply:
Happy to change if you want?